xray/plot/plot.py

```diff
@@ -55,6 +55,17 @@ def _load_default_cmap(fname='default_colormap.csv'):
     return LinearSegmentedColormap.from_list('viridis', cm_data)
 
 
+def _title_for_slice(darray):
+    '''
+    If the dataarray comes from a slice we can show that info in the title
+    '''
+    title = []
+    for key, value in darray.coords.items():
+        if value.size == 1:
```

Comment:
Yes. Here's a 4d array with a single size 1 coordinate:
```
  * dim_0    (dim_0) int64 0 1 2
  * dim_1    (dim_1) int64 0 1 2 3
  * dim_2    (dim_2) int64 0 1 2 3 4 5 6 7 8 9
  * dim_3    (dim_3) int64 0

In [11]: _title_for_slice(b)
Out[11]: 'dim_3 = [0]'

In [13]: _title_for_slice(b.isel(dim_1=2))
Out[13]: 'dim_1 = 2, dim_3 = [0]'
```
It shows square brackets around the value, since it's an array in this case.

However, this won't come up in the current implementation since we don't allow 2d plots to work on say a 2 x 4 x 1 array:

```
In [14]: b.isel(dim_1=2).plot.contourf()

...

ValueError: contourf plots are for 2 dimensional DataArrays. Passed DataArray has 3 dimensions
```

I've thought about making this a special case and allowing it to plot. Thoughts?